x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicyTests.java

```diff
@@ -305,7 +317,7 @@ protected LifecyclePolicy mutateInstance(LifecyclePolicy instance) {
             }
             default -> throw new AssertionError("Illegal randomisation branch");
         }
-        return new LifecyclePolicy(TimeseriesLifecycleType.INSTANCE, name, phases, randomMeta());
+        return new LifecyclePolicy(TimeseriesLifecycleType.INSTANCE, name, phases, randomMeta(), randomBoolean());
```

Comment:
I think this method needs to actually mutate the deprecation flag in the `switch` statement to ensure the hashCode/equals methods are tested correctly.